src/flow_solver/flow_solver_cases/periodic_turbulence.cpp

```diff
@@ -295,80 +295,109 @@ double PeriodicTurbulence<dim, nstate>::get_deviatoric_strain_rate_tensor_based_
 }
 
 template<int dim, int nstate>
-double PeriodicTurbulence<dim, nstate>::get_numerical_entropy(const std::shared_ptr <DGBase<dim, double>> dg) const
+double PeriodicTurbulence<dim, nstate>::get_numerical_entropy(
+        const std::shared_ptr <DGBase<dim, double>> dg
+        ) const
 {
     const double poly_degree = this->all_param.flow_solver_param.poly_degree;
-    dealii::LinearAlgebra::distributed::Vector<double> mass_matrix_times_solution(dg->right_hand_side);
-    if(this->all_param.use_inverse_mass_on_the_fly){
-            dg->apply_global_mass_matrix(dg->solution,mass_matrix_times_solution);
-    } else {
-        dg->global_mass_matrix.vmult( mass_matrix_times_solution, dg->solution);
-    }
 
     const unsigned int n_dofs_cell = dg->fe_collection[poly_degree].dofs_per_cell;
     const unsigned int n_quad_pts = dg->volume_quadrature_collection[poly_degree].size();
     const unsigned int n_shape_fns = n_dofs_cell / nstate;
-    // We have to project the vector of entropy variables because the mass matrix has an interpolation from solution nodes built into it.
+
     OPERATOR::vol_projection_operator<dim,2*dim> vol_projection(1, poly_degree, dg->max_grid_degree);
     vol_projection.build_1D_volume_operator(dg->oneD_fe_collection_1state[poly_degree], dg->oneD_quadrature_collection[poly_degree]);
 
+    // Construct the basis functions and mapping shape functions.
     OPERATOR::basis_functions<dim,2*dim> soln_basis(1, poly_degree, dg->max_grid_degree); 
     soln_basis.build_1D_volume_operator(dg->oneD_fe_collection_1state[poly_degree], dg->oneD_quadrature_collection[poly_degree]);
 
-    dealii::LinearAlgebra::distributed::Vector<double> entropy_var_hat_global(dg->right_hand_side);
+    OPERATOR::mapping_shape_functions<dim,2*dim> mapping_basis(1, poly_degree, dg->max_grid_degree);
+    mapping_basis.build_1D_shape_functions_at_grid_nodes(dg->high_order_grid->oneD_fe_system, dg->high_order_grid->oneD_grid_nodes);
+    mapping_basis.build_1D_shape_functions_at_flux_nodes(dg->high_order_grid->oneD_fe_system, dg->oneD_quadrature_collection[poly_degree], dg->oneD_face_quadrature);
+
     std::vector<dealii::types::global_dof_index> dofs_indices (n_dofs_cell);
+    
+    double integrand_numerical_entropy_function=0;
+    double integral_numerical_entropy_function=0;
+    const std::vector<double> &quad_weights = dg->volume_quadrature_collection[poly_degree].get_weights();
+    std::shared_ptr < Physics::Euler<dim, nstate, double > > euler_physics = std::dynamic_pointer_cast<Physics::Euler<dim,dim+2,double>>(PHiLiP::Physics::PhysicsFactory<dim,nstate,double>::create_Physics(dg->all_parameters));
```

Comment:
Is `euler_physics` being used somewhere?

Reply:
Good point, I think that is there because I copied code from elsewhere. I'll remove it as `navier_stokes_physics` is used instead.